### PR TITLE
Fixes #6: Refactor literal [^0-9.] in Battle.java to reduce its duplication

### LINKED ISSUE #6: Refactor literal "[^0-9.]" in Battle.java to reduce its duplication
### Issue Type: Technical Debt

[SonarCloud Issue ](https://sonarcloud.io/project/issues?fileUuids=AZIuy3lKSndwZUJusoqZ&issueStatuses=OPEN%2CCONFIRMED&id=soen6431group10_robocodeFall2024&open=AZIuy4cUSndwZUJuspPu&tab=code)
File: `robocode.battle/src/main/java/net/sf/robocode/battle/Battle.java`
Literal: `[^0-9.]`
Number of times duplicated: 3 
SonarQube Rule: [java - String literals should not be duplicated](https://sonarcloud.io/organizations/soen6431group10/rules?open=java%3AS1192&rule_key=java%3AS1192)

### Description

The literal `[^0-9.]` is duplicated in three times, which could make the process of refactoring complex and error-prone, as any change would need to be propagated on all occurrences.

### Impact on Software Qualities

Maintainability: Duplicating literals in code reduces maintainability by increasing the risk of errors, complicating refactoring, and hindering readability and collaboration.

### Proposed Solution
Define a new method `stripNonNumericCharacters()` to replace all repeated `[^0-9.]` statement. The method can be invoked from many places, but only need to be updated in a single place.

### Acceptance Criteria

- All instances of the literal `[^0-9.]` in the file are replaced with the constant named `invalidCharacters` .

### Steps to Reproduce

1. Run a SonarQube analysis on the project.
2. Observe the Code Smell report for `[^0-9.]` duplicated literal.
3. Define a new method which return the desired string of those duplicated codes.

### PR DESCRIPTION
**Pull request by LeeELIOT**
# Refactor literal [^0-9.] in Battle.java to reduce its duplication
[GitHub Issue Link](https://github.com/soen6431group10/robocodeFall2024/issues/6)
[SonarCloud Issue Link](https://sonarcloud.io/project/issues?fileUuids=AZIuy3lKSndwZUJusoqZ&issueStatuses=OPEN%2CCONFIRMED&id=soen6431group10_robocodeFall2024&open=AZIuy4cUSndwZUJuspPu&tab=code)

The pull request refactors the literal `[^0-9.]` in Battle.java by extracting a method named `stripNonNumericCharacters` in replace of all the same `replaceAll("[^0-9.]", "")` operation. This change aims to avoid duplication and improve project's readability and maintainability.

## Key Changes:

- **`stripNonNumericCharacters()`**: Define a new method to perform all the same string replacement.
- **`[^0-9.]`**: Remove all the redundant and identical string.

## Benefits:

- **Reduced Cognitive Load**: Extracting method to make code less repeated, which could lead to easier maintainence. 
- **Improved Maintainability**: Isolating specific functionality into well-named methods allows for easier updates and modifications in the future.
- **Adherence to Single Responsibility Principle**: Each method is given a distinct, clear responsibility, leading to better organization of the code.

## Conclusion:

This refacoring successfully address duplicated problem of [^0-9.], which improves maintainability and readability.

### CODE REVIEW

File: robocode.battle/src/main/java/net/sf/robocode/battle/Battle.java [hunk 1]
```diff
@@ -727,17 +728,17 @@ private void computeInitialPositions(String initialPositions) {
 
 			if (len >= 1 && coords[0].trim().length() > 0) {
 				try {
-					x = Double.parseDouble(coords[0].replaceAll("[^0-9.]", ""));
+					x = Double.parseDouble(coords[0].replaceAll(invalidCharacters, ""));
```
Comment (0matleb2) [inline robocode.battle/src/main/java/net/sf/robocode/battle/Battle.java:731]: While this resolves the SQ issue by reusing the regex string, another solution I'd suggest is encapsulating the string replacement behavior into a well name method and reusing that.
```java
private String stripNonNumericCharacters(String input) {
    return input.replaceAll("[^0-9.]", "");
}
```
```suggestion
					x = Double.parseDouble(stripNonNumericCharacters(coords[0]));
```

File: .idea/kotlinc.xml
Comment (0matleb2) [inline .idea/kotlinc.xml:1]: Can you revert these .idea configuration file changes please?

File: robocode.battle/src/main/java/net/sf/robocode/battle/Battle.java (hunk 1 above)
Comment (0matleb2) [inline robocode.battle/src/main/java/net/sf/robocode/battle/Battle.java:731]: While this resolves the SQ issue by reusing the regex string, another solution I'd suggest is encapsulating the string replacement behavior into a well name method and reusing that.
```java
private String stripNonNumericCharacters(String input) {
    return input.replaceAll("[^0-9.]", "");
}
```
```java
x = Double.parseDouble(stripNonNumericCharacters(coords[0]));
```

**Review by 0matleb2 — CHANGES_REQUESTED** [review]

File: .idea/kotlinc.xml
Comment (LeeELIOT) [reply .idea/kotlinc.xml:1]: Sure!

**Review by 0matleb2 — APPROVED** [review]
Thanks for the changes! Approved.

**Review by Shine-23 — APPROVED** [review]
Approved !

**Review by sakshirk — APPROVED** [review]
LGTM!

**Pull request merged**